README.md

```diff
@@ -1,5 +1,146 @@
+
+````
+class StackTest {
+    @Test
+    public void topIsLastPushedValue() {
+        fail("Not implemented");
+    }
+}
+````
+
+> Example 3: Skeletonized code that is provided to students.
+
+We can see that the only code that remains inside the function is that of the `REPLACE-WITH` marker. This gives us the main usage that `repobee-santizer` was developed for, it allows us to combine finished solutions with the "skeletonized" solutions that are provided to students.
+
+# Prefixing
+Sometimes (usually) we want code that can run, its a good thing then that `repobee-sanitizer` blocks can be commented out! Example two will produce the same output as the following:
+
+````
+class StackTest {
+    @Test
+    public void topIsLastPushedValue() {
+//REPOBEE-SANITIZER-BLOCK
+        // Arrange
+        int value = 1338;
+
+        // Act
+        emptyStack.push(value);
+        stack.push(value);
+
+        int emptyStackTop = emptyStack.top();
+        int stackTop = stack.top();
+
+        // Assert
+        assertThat(emptyStackTop, equalTo(value));
+        assertThat(stackTop, equalTo(value));
+//REPOBEE-SANITIZER-REPLACE-WITH
+//        fail("Not implemented");
+//REPOBEE-SANITIZER-END
+    }
+}
+````
+> Example 4: Java code with `repobee-sanitizer` related syntax commented out 
+
+`repobee-sanitizer` automatically detects if there is a prefix in front of any markers. This way we can have java comments: `//`, python comments: `#` or similar preceding our markers. **This means code can still compile!**
+
+## Some rules for prefixing:
+
+`repobee-sanitizer`:
+* Determines prefix as any text that comes before `REPOBEE-SANITIZER-BLOCK`
+* Only determines prefix on a block-to-block basis, meaning that the prefix selected at a `BLOCK` marker must be used untill and including the next `END` marker 
+    * This means that all `repobee-sanitizer` blocks can have individuall prefixes
+
+
+# Syntax
+for `repobee-sanitizer` to work, marker syntax must be correct, this includes spelling of the markers themselves, the markers currently are as follows:
+
+- REPOBEE-SANITIZER-BLOCK
+    - REQUIRED: A block is not a block without a start marker
+    - Indicates the start of a block. Any text will be removed untill reaching a `REPLACE-WITH` or `END` marker. 
+- REPOBEE-SANITIZER-REPLACE-WITH
+    - OPTIONAL: but requires a start and end block.
+    - Any text between this marker and the next `END` marker will remain.
+- REPOBEE-SANITIZER-END
+    - REQUIRED: Must exist for each start block
+    - Indicates the end of a block.
+
+If a marker is incorrectly spelled, `repobee-sanitizer` will report an error. 
 
 # License
 See [LICENSE](LICENSE) for details.
+
+# Terminology
+Block
+Marker
```

Comment:
```suggestion
```
This is redundant for now, terminology may change as we move along and this section doesn't explain anything anyway. Remove.